src/Kit/System/Watchdog/Supervisor.h

```diff
@@ -0,0 +1,167 @@
+    The class uses a composition relationship with SList<WatchedThread> to manage
+    the collection of monitored threads.
+ */
+class Supervisor
+{
+public:
+    /** Enables the watchdog system. Must be called before any threads
+        can be monitored.
+
+        @return true if watchdog was successfully enabled, false otherwise
+     */
+    static bool enableWdog() noexcept
+    {
+        return Kit_System_Watchdog_hal_enable_wdog();
+    }
+
+    /** Triggers the watchdog to reset the system. This is called when
+        a monitored thread has failed to check in within its timeout period.
+     */
+    static void tripWdog() noexcept
+    {
+        Kit_System_Watchdog_hal_trip_wdog();
+    }
+
+    /** Begins monitoring a thread. The thread will be added to the list
+        of monitored threads. Once a thread has been added to the Supervisor,
+        this method can NOT be called again until after endWatching() has
+        been called on the same thread.
+        @param threadToMonitor Reference to the WatchedThread to begin monitoring
+     */
+    static void beginWatching( WatchedThread& threadToMonitor ) noexcept;
+
+    /** Stops monitoring a thread. The thread will be removed from the list
+        of monitored threads.
+        @param threadBeingMonitored Reference to the WatchedThread to stop monitoring
+     */
+    static void endWatching( WatchedThread& threadBeingMonitored ) noexcept;
+
+    /** Monitors all registered threads. This method should be called periodically
+        (typically every tick/event cycle) by the supervisor thread. It includes
+        a tick divider to reduce the performance impact of being called frequently.
+
+        This method checks each monitored thread's currentCountMs against its
+        wdogTimeoutMs and triggers the watchdog if any thread has timed out.
+     */
+    static void monitorThreads() noexcept;
+
+    /** Reloads/resets the watchdog counter for a specific thread. This is
+        called by the thread to indicate it is still alive and processing.
+
+        @param thread Reference to the WatchedThread to reload
+     */
+    static void reloadThread( WatchedThread& thread ) noexcept;
+
+    /** Kicks the hardware watchdog. Should be called periodically when all
+        monitored threads are healthy. Typically called by the supervisor thread.
+     */
+    static void kickWdog() noexcept
+    {
+        Kit_System_Watchdog_hal_kick_wdog();
+    }
+
+protected:
+    /// List of threads being monitored
+    static Kit::Container::SList<WatchedThread> m_watchedThreads;
+
+    /// Mutex for thread-safe access to the watched threads list
+    static Mutex m_mutex;
+
+    /// Current tick count for the divider
+    static uint32_t m_currentTick;
+
+    /// Flag indicating if watchdog system is enabled
+    static bool m_isEnabled;
+
+    /// Time marker for elapsed time calculations
+    static uint32_t m_timeMarker;
```

Comment:
Re-order the data member so that the "larger sized" (with respect to address alignment) are first - this helps reduce/eliminate pad bytes in the class
```suggestion
    /// List of threads being monitored
    static Kit::Container::SList<WatchedThread> m_watchedThreads;

    /// Mutex for thread-safe access to the watched threads list
    static Mutex m_mutex;

    /// Current tick count for the divider
    static uint32_t m_currentTick;

    /// Time marker for elapsed time calculations
    static uint32_t m_timeMarker;

        /// Flag indicating if watchdog system is enabled
    static bool m_isEnabled;
```